Internal error: `NoSuchElementException` in `TrailingCommaRule` when using `--experimental`

<!-- The bug you're experiencing might have already be reported!
Please search in the [issues](https://github.com/pinterest/ktlint/issues) before creating one. -->

## Expected Behavior
<!---Tell us what should happen -->
<!-- If possible provide a link to official code style guide -->
ktlint doesn't crash with an internal error.

## Observed Behavior
<!---Tell us what happens instead of the expected behavior -->
ktlint crashes with an internal error when running on a specific file due to a `NoSuchElementException` in `TrailingCommaRule` when using `--experimental`.

## Steps to Reproduce
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
1. In an empty folder, create a file `Test.kt` with the following contents:

```kotlin
class Test {
  var foo = Bar()
    set(value) {
    }
}
```

2. Run `ktlint --experimental --debug Test.kt`.
3. ktlint produces an output similar to the following:

<details>

```plaintext
[DEBUG] Discovered ruleset with " standard" id.
[DEBUG] Discovered ruleset with "experimental" id.
[DEBUG] Discovered reporter with "baseline" id.
[DEBUG] Discovered reporter with "checkstyle" id.
[DEBUG] Discovered reporter with "json" id.
[DEBUG] Discovered reporter with "html" id.
[DEBUG] Discovered reporter with "plain" id.
[DEBUG] Discovered reporter with "sarif" id.
[DEBUG] Initializing "plain" reporter with {verbose=false, color=false, color_name=DARK_GRAY}
[DEBUG] Checking D:\tmp\Test.kt
Resolving .editorconfig files for D:\tmp\Test.kt file path
Loaded .editorconfig: []
[DEBUG] Internal Error (experimental:trailing-comma)
com.pinterest.ktlint.core.RuleExecutionException: java.util.NoSuchElementException: Sequence contains no 
element matching the predicate.
        at com.pinterest.ktlint.core.KtLint$lint$1.invoke(KtLint.kt:166)
        at com.pinterest.ktlint.core.KtLint$lint$1.invoke(KtLint.kt:147)
        at com.pinterest.ktlint.core.KtLint$visitor$2$3.invoke(KtLint.kt:317)
        at com.pinterest.ktlint.core.KtLint$visitor$2$3.invoke(KtLint.kt:315)
        at com.pinterest.ktlint.core.ast.PackageKt.visit(package.kt:229)
        at com.pinterest.ktlint.core.ast.PackageKt.visit(package.kt:230)
        at com.pinterest.ktlint.core.ast.PackageKt.visit(package.kt:230)
        at com.pinterest.ktlint.core.ast.PackageKt.visit(package.kt:230)
        at com.pinterest.ktlint.core.ast.PackageKt.visit(package.kt:230)
        at com.pinterest.ktlint.core.ast.PackageKt.visit(package.kt:230)
        at com.pinterest.ktlint.core.KtLint$visitor$2.invoke(KtLint.kt:315)
        at com.pinterest.ktlint.core.KtLint$visitor$2.invoke(KtLint.kt:293)
        at com.pinterest.ktlint.core.KtLint.lint(KtLint.kt:147)
        at com.pinterest.ktlint.internal.FileUtilsKt.lintFile(FileUtils.kt:172)
        at com.pinterest.ktlint.KtlintCommandLine.process(Main.kt:395)
        at com.pinterest.ktlint.KtlintCommandLine.access$process(Main.kt:89)
        at com.pinterest.ktlint.KtlintCommandLine$lintFiles$3.invoke$lambda-0(Main.kt:289)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)     
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)     
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.util.NoSuchElementException: Sequence contains no element matching the predicate.        
        at com.pinterest.ktlint.ruleset.experimental.trailingcomma.TrailingCommaRule.visitValueList(TrailingCommaRule.kt:357)
        at com.pinterest.ktlint.ruleset.experimental.trailingcomma.TrailingCommaRule.visit(TrailingCommaRule.kt:61)
        at com.pinterest.ktlint.core.KtLint$lint$1.invoke(KtLint.kt:154)
        ... 20 more
D:\tmp\Test.kt:3:9: Internal Error (experimental:trailing-comma). Please create a ticket at https://github.com/pinterest/ktlint/issues (if possible, please re-run with the --debug flag to get the stacktrace and provide the source code that triggered an error) (cannot be auto-corrected)
[DEBUG] 1625ms / 1 file(s) / 1 error(s)
```
</details>

## Additional Information
I was just migrating my project ltex-ls to use the new `trailing-comma` rules when I discovered this. However, the bug occurs even if the `trailing-comma` rules have not been enabled (no `.editorconfig` file or `trailing-comma` rules explicitly set to `false`). It suffices to use `--experimental` to trigger the bug.

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Version of ktlint used: 0.43.0
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): Command-line ktlint
* Version of Gradle used (if applicable): N/A
* Operating System and version: Windows 10
* Link to your project (if it's a public repository): https://github.com/valentjn/ltex-ls (but you obviously won't find the example code there)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,7 @@ This project adheres to [Semantic Versioning](https://semver.org/).
 
 ### Fixed
 - Fix false positive in rule spacing-between-declarations-with-annotations ([#1281](https://github.com/pinterest/ktlint/issues/1281))
+- Fix NoSuchElementException for property accessor (`trailing-comma`) ([#1280](https://github.com/pinterest/ktlint/issues/1280))
 
 ### Changed
 - Update Kotlin version to `1.6.0` release

--- a/ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/trailingcomma/TrailingCommaRule.kt
+++ b/ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/trailingcomma/TrailingCommaRule.kt
@@ -129,10 +129,12 @@ public class TrailingCommaRule :
         autoCorrect: Boolean
     ) {
         if (node.treeParent.elementType != ElementType.FUNCTION_LITERAL) {
-            val inspectNode = node
+            node
                 .children()
-                .last { it.elementType == ElementType.RPAR }
-            node.reportAndCorrectTrailingCommaNodeBefore(inspectNode, emit, autoCorrect)
+                .lastOrNull { it.elementType == ElementType.RPAR }
+                ?.let {
+                    node.reportAndCorrectTrailingCommaNodeBefore(it, emit, autoCorrect)
+                }
         }
     }
 

--- a/ktlint-ruleset-experimental/src/test/kotlin/com/pinterest/ktlint/ruleset/experimental/TrailingCommaRuleTest.kt
+++ b/ktlint-ruleset-experimental/src/test/kotlin/com/pinterest/ktlint/ruleset/experimental/TrailingCommaRuleTest.kt
@@ -908,6 +908,26 @@ class TrailingCommaRuleTest {
             .isEqualTo(autoCorrectedCode)
     }
 
+    @Test
+    fun `Trailing comma is not added for property setter`() {
+        val code =
+            """
+            class Test {
+              var foo = Bar()
+                set(value) {
+                }
+            }
+            """.trimIndent()
+
+        val editorConfigFilePath = writeEditorConfigFile(
+            ALLOW_TRAILING_COMMA_ON_DECLARATION_SITE, ALLOW_TRAILING_COMMA_ON_CALL_SITE
+        ).absolutePath
+
+        assertThat(TrailingCommaRule().lint(editorConfigFilePath, code)).isEmpty()
+        assertThat(TrailingCommaRule().format(editorConfigFilePath, code))
+            .isEqualTo(code)
+    }
+
     private fun writeEditorConfigFile(vararg editorConfigProperties: Pair<PropertyType<Boolean>, String>) = editorConfigTestRule
         .writeToEditorConfig(
             mapOf(*editorConfigProperties)

